index.js

```diff
@@ -1,14 +1,27 @@
-const { curryN } = require('crocks')
+const { curryN }         = require('crocks')
+const { flip, is, when } = require('ramda')
+
+const addReqData = (err, req) =>
+  when(is(Error), flip(Object.assign)({
+    action     : req.pathname,
+    component  : 'paperplane',
+    httpMethod : req.method,
+    params     : req.params,
+    ua         : req.headers['user-agent'],
+    url        : req.protocol + '://' + req.headers.host + req.url
+  }))(err)
 
 const handle = curryN(3, (airbrake, req, err) => {
-  err.action     = req.pathname
-  err.component  = 'paperplane'
-  err.httpMethod = req.method
-  err.params     = req.params
-  err.ua         = req.headers['user-agent']
-  err.url        = req.protocol + '://' + req.headers.host + req.url
-  airbrake.notify(err)
-  return Promise.reject(err)
+  const error = addReqData(err, req)
```

Comment:
This `error` does not need to be assigned, since we're mutating the `err` object directly (because you can't modify them immutably 😢 ).  So just `addReqData(err, req)` instead.